File: cpp/563_BinaryTreeTilt.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int tilt = 0;
    int mutateTilt(TreeNode* root) {
        if (!root) return 0;
        
        int left = mutateTilt(root->left);
        int right = mutateTilt(root->right);
        
        tilt += abs(left - right);
        
        return left + right + root->val;
    }
    
    int findTilt(TreeNode* root) {
        tilt = 0;
        mutateTilt(root);
        return tilt;
    }
};

File: python/563_BinaryTreeTilt.py
class Solution:
    def findTilt(self, root: TreeNode) -> int:
        def sumTree(node):
            if not node: return 0
            
            left = sumTree(node.left)
            right = sumTree(node.right)
            
            self.tilt += abs(left - right)
            
            return left + right + node.val
        
        self.tilt = 0
        sumTree(root)
        return self.tilt
